src/test/scala/chiselTests/BundleElementsSpec.scala

```diff
@@ -207,7 +207,44 @@ class UsesBundleWithGeneratorField extends Module {
 
   out := DontCare
 
-  assertElementsMatchExpected(out)("superQux" -> _.superQux, "superFoo" -> _.superFoo)
+  assertElementsMatchExpected(out)(
+    "superFoo" -> _.superFoo,
+    "superQux" -> _.superQux
+  )
+}
+
+
```

Comment:
seems like the cases we want to see are cross products of: 
a. vals in the constructor args
b. vals in the body
c. vals in superclass/traits bodies

We should have a clear story on the order of these and the order within each of a, b, c and then show that as expected they go in order with a few test cases.

Reply:
before the PR, i think that `b` had expected behavior that we don't want to change in this PR except for weird interactions flagged by the bug